pubsub/subscriber.go

```diff
@@ -31,17 +31,19 @@ type Subscriber struct {
 	handlers  map[string]Handler
 }
 
-func NewSubscriber(thor *thorclient.Client, db *influxdb.DB, blockChan chan *Block) (*Subscriber, error) {
-	genesis, err := thor.Block("0")
+func NewSubscriber(thorURL string, db *influxdb.DB, blockChan chan *Block) (*Subscriber, error) {
+	tclient := thorclient.New(thorURL)
+
+	genesis, err := tclient.Block("0")
 	if err != nil {
 		slog.Error("failed to get genesis block", "error", err)
 		return nil, err
 	}
 	chainTag := fmt.Sprintf("%d", genesis.ID[31])
 
-	liveness := liveness2.New(thor)
-	poa := authority.NewList(thor)
-	hayabusa, err := pos.NewStaker(thor)
+	liveness := liveness2.New(thorclient.New(thorURL))
```

Comment:
since you are declaring `tsclient` above, can it be used here and in the 2 occurrences below?